corehq/apps/locations/permissions.py

```diff
@@ -321,3 +321,18 @@ def _inner(request, domain, loc_id, *args, **kwargs):
         return location_restricted_response(request)
 
     return require_can_edit_or_view_locations(_inner)
+
+
+def can_access_location_data(view_fn):
+    """
+    Decorator controlling a user's access to VIEW data for a specific location.
+    """
+    @wraps(view_fn)
+    def _inner(request, domain, *args, **kwargs):
+        loc_id = request.GET.get('location_id')
+        def call_view(): return view_fn(request, domain, *args, **kwargs)
+        if loc_id is not None and not user_can_access_location_id(domain, request.couch_user, loc_id):
+            return HttpResponse(status=403)
```

Comment:
nit: maybe also add a message indicating they are getting this response due to location restrictions? would at least be helpful for debugging

Reply:
> nit: maybe also add a message indicating they are getting this response due to location restrictions? would at least be helpful for debugging

loc_id is being used in couple of places. So made it a variable. Also added the info message

Reply:
You could also return `location_restricted_response` or raise `location_restricted_exception`